src/Microsoft.DotNet.ProjectModel/DependencyContextBuilder.cs

```diff
@@ -5,38 +5,72 @@
 using Microsoft.DotNet.ProjectModel;
 using Microsoft.DotNet.ProjectModel.Compilation;
 using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Frameworks;
 
 namespace Microsoft.Extensions.DependencyModel
 {
     public static class DependencyContextBuilder
     {
-        public static DependencyContext FromLibraryExporter(LibraryExporter libraryExporter, string target, string runtime)
+        public static DependencyContext Build(CommonCompilerOptions compilerOptions, LibraryExporter libraryExporter, string configuration, NuGetFramework target, string runtime)
         {
             var dependencies = libraryExporter.GetAllExports();
 
-            return new DependencyContext(target, runtime,
-                GetLibraries(dependencies, export => export.CompilationAssemblies),
-                GetLibraries(dependencies, export => export.RuntimeAssemblies));
+            return new DependencyContext(target.DotNetFrameworkName, runtime,
+                GetCompilationOptions(compilerOptions),
+                GetLibraries(dependencies, target, configuration, export => export.CompilationAssemblies),
+                GetLibraries(dependencies, target, configuration, export => export.RuntimeAssemblies));
         }
 
-        private static Library[] GetLibraries(IEnumerable<LibraryExport> dependencies, Func<LibraryExport, IEnumerable<LibraryAsset>> assemblySelector)
+        private static CompilationOptions GetCompilationOptions(CommonCompilerOptions compilerOptions)
         {
-            return dependencies.Select(export => GetLibrary(export, assemblySelector(export), dependencies)).ToArray();
+            return new CompilationOptions(compilerOptions.Defines,
+                compilerOptions.LanguageVersion,
+                compilerOptions.Platform,
+                compilerOptions.AllowUnsafe,
+                compilerOptions.WarningsAsErrors,
+                compilerOptions.Optimize,
+                compilerOptions.KeyFile,
+                compilerOptions.DelaySign,
+                compilerOptions.PublicSign,
+                compilerOptions.EmitEntryPoint);
         }
 
-        private static Library GetLibrary(LibraryExport export, IEnumerable<LibraryAsset> libraryAssets, IEnumerable<LibraryExport> dependencies)
+        private static Library[] GetLibraries(IEnumerable<LibraryExport> dependencies, NuGetFramework target, string configuration, Func<LibraryExport, IEnumerable<LibraryAsset>> assemblySelector)
         {
+            return dependencies.Select(export => GetLibrary(export, target, configuration, assemblySelector(export), dependencies)).ToArray();
```

Comment:
This looks really inefficient. Couldn't this be done with a single foreach loop over all dependencies?